File: api_commandes/src/main/java/fr/univamu/iut/api_commandes/commande/Commande.java
package fr.univamu.iut.api_commandes.commande;

import jakarta.ws.rs.Path;

import java.util.Date;

public class Commande {

    protected int id_commande   ;

    protected int id_user;

    protected int prix;

    protected Date date_retrait;

    protected String status;

    protected String relais;


    public Commande() {}

    public Commande (int id_commande, int id_user, int prix, Date date_retrait, String status, String relais ) {
        this.id_commande = id_commande;
        this.id_user = id_user;
        this.prix = prix;
        this.date_retrait = date_retrait;
        this.status = status;
        this.relais = relais;
    }

    public int getId_commande()
    {
        return id_commande;
    }
    public void setId_commande(int id_commande)
    {
        this.id_commande = id_commande;
    }

    public int getId_user()
    {
        return id_user;
    }
    public void setId_user(int id_user)
    {
        this.id_user = id_user;
    }

    public int getPrix()
    {
        return prix;
    }
    public void setPrix(int prix)
    {
        this.prix = prix;
    }

    public java.sql.Date getDate_retrait()
    {
        return (java.sql.Date) date_retrait;
    }
    public void setDate_retrait(Date date_retrait)
    {
        this.date_retrait = date_retrait;
    }

    public String getStatus()
    {
        return status;
    }
    public void setStatus(String relais)
    {
        this.status = status;
    }

    public String getRelais()
    {
        return relais;
    }
    public void setRelais(String relais)
    {
        this.relais = relais;
    }
}

File: domain/Commande.php
<?php

namespace domain;

/**
 * Classe représentant une commande
 * */
class Commande
{
    /**
     * Identifiant d'une commande
     * */
    private int $idCommande;

    /**
     * Classe représentant une commande
     * */
    private int $idUser;

    /**
     * Prix d'une commande
     * */
    private float $prix;

    /**
     * Date de retrait de la commande
     * */
    private int $dateRetrait;

    /**
     * Status de la commande
     * */
    private string $status;

    /**
     * Relais une commande
     * */
    private string $relais;

    /**
     * Constructeur de la classe
     * @param int $idCommande
     * @param int $idUser
     * @param int $prix
     * @param int $dateRetrait
     * @
     * */
    public function __construct(int $idUser, int $idCommande, int $prix, int $dateRetrait, string $status, string $relais)
    {
        $this->idCommande = $idCommande;
        $this->idUser = $idUser;
        $this->prix = $prix;
        $this->dateRetrait = $dateRetrait;
        $this->status = $status;
        $this->relais = $relais;
    }

    /**
     * @return int
     */
    public function getIdCommande(): int
    {
        return $this->idCommande;
    }

    /**
     * @return int
     */
    public function getIdUser(): int
    {
        return $this->idUser;
    }

    /**
     * @return float
     */
    public function getPrix(): float
    {
        return $this->prix;
    }

    /**
     * @return int
     */
    public function getDateRetrait(): int
    {
        return $this->dateRetrait;
    }

    /**
     * @return string
     */
    public function getRelais(): string
    {
        return $this->relais;
    }

    /**
     * @return string
     */
    public function getStatus(): string
    {
        return $this->status;
    }
}